public/app/features/dashboard/state/DashboardMigrator.ts

```diff
@@ -793,7 +793,10 @@ export class DashboardMigrator {
 
     if (oldVersion < 37) {
       panelUpgrades.push((panel: PanelModel) => {
-        if (panel.options?.legend && panel.options.legend.displayMode === 'hidden') {
+        if (
+          panel.options?.legend &&
+          (panel.options.legend.displayMode === 'hidden' || panel.options.legend.showLegend === false)
```

Comment:
Yeah, looks okay! Should we add a comment above the if statement explaining the supported/not supported situation? Your PR description is good enough, but just to have it in here since `if (showLegend === false) showLegend = false` is definitely confusing

Reply:
I totally agree! Done! Thanks for the suggestion!